### PR TITLE
rtss: Fix `pre_install` and `persist`

### LINKED ISSUE #9789: [Bug]: RivaTuner Statistics Server / rtss - not persisting all user configs
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.
- [X] I have verified that I am using the latest version of Scoop and corresponding bucket.

### Package Name

rtss

### Expected/Current Behaviour

"Regular" RTSS keeps these folders and files on uninstall because they contain user-made changes:

- \Profiles
- \Plugins\Client\*.cfg
- \Plugins\Client\Overlays\*.ovl

scoop package only persists this folder:

- \scoop\persist\rtss\Profiles
- \scoop\persist\rtss\Plugins\Client\Overlays

Note that `\Plugins\Client\*.cfg` are missing. This means that any plugin configurations added or configured by user will be reset on update & deleted on cleanup.

It would be expected to at least persist files in this manner:
`\Plugins\Client\*.cfg`

### Steps to Reproduce

```console
scoop install rtss

Open RTSS and enable and configure plugins offered, eg I've added and enabled custom overlay and hotkey to toggle it.

When new version is released:
scoop update rtss

Customizations will be gone.
```

### Possible Solution

This is related to the issue I've just reported for MSI Afterburner ( https://github.com/ScoopInstaller/Extras/issues/9788 )

Either persist files `\Plugins\Client\*.cfg` or add a Note for user to move the files manually before cleanup deletes them.

### Scoop and Buckets Version

```console
Current Scoop version:
v0.3.1 - Released at 2022-11-15

'extras' bucket:
b4ebfb2dc (HEAD -> master, origin/master, origin/HEAD) jackett: Update to version 0.20.2267

'java' bucket:
fec2bd1a (HEAD -> master, origin/master, origin/HEAD) temurin8-nightly-jre: Update to version 8.0.362-3.0.202211181811

'main' bucket:
e13981012 (HEAD -> master, origin/master, origin/HEAD) tflint: Update to version 0.43.0
```

### Scoop Config

```console
last_update          scoop_repo                              scoop_branch
-----------          ----------                              ------------
19.11.2022. 16:05:54 https://github.com/ScoopInstaller/Scoop master
```

### PowerShell Version

```console
Name                           Value
----                           -----
PSVersion                      7.3.0
PSEdition                      Core
GitCommitId                    7.3.0
OS                             Microsoft Windows 10.0.22621
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Additional Softwares

_No response_

### PR DESCRIPTION
**Pull request by Zliced13**
Closes #9789

- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by HUMORCE** [comment]
consider using (un)install scripts instead of `persist` function for persist these:

```
Plugins\Client\*.cfg
Plugins64\Client\*.cfg
```

i dont really know what will be created in which dir, maybe:
```
Plugins\*.cfg -Exclude 'VCEAMFPreset0.cfg', 'VCEAMFPreset1.cfg', 'VCEAMFPreset2.cfg', 'VCEPreset0.cfg', 'VCEPreset1.cfg', 'VCEPreset2.cfg'
Plugins64\*.cfg -Exclude <...>
```

**Comment by Zliced13** [comment]
What is the command to persist something in Scoop JSON manifests?

**Comment by HUMORCE** [comment]
refer `pre_(un)install` of this manifest: [everything.json](https://github.com/ScoopInstaller/Extras/blob/master/bucket/everything.json)

https://github.com/ScoopInstaller/Extras/blob/6889083f9c3c00a6366e41940fe0e5507347fb61/bucket/everything.json#L23

https://github.com/ScoopInstaller/Extras/blob/6889083f9c3c00a6366e41940fe0e5507347fb61/bucket/everything.json#L49

**Comment by Zliced13** [comment]
Why not use the 

> consider using (un)install scripts instead of `persist` function for persist these:
> 
> ```
> Plugins\Client\*.cfg
> Plugins\Client\Overlays\*.ovl -Exclude 'benchmark.ovl', 'classic.ovl', 'sample.ovl'
> Plugins64\Client\*.cfg
> ```
> 
> i dont really know what will be created in which dir, maybe:
> 
> ```
> Plugins\*.cfg -Exclude 'VCEAMFPreset0.cfg', 'VCEAMFPreset1.cfg', 'VCEAMFPreset2.cfg', 'VCEPreset0.cfg', 'VCEPreset1.cfg', 'VCEPreset2.cfg'
> Plugins64\*.cfg -Exclude <...>
> 
> if ‘Plugins64\Client\Overlays’ exist:
> Plugins64\Client\Overlays\*.ovl
> ```

Why use this, instead of using the `persist` function?

**Comment by HUMORCE** [comment]
in this case, HotkeyHandler.cfg` and `OverlayEditor.cfg` are known config files(user reported), but we dont know the filenames of other config files, and we cannot persist entire dir for `Plugins\Client`.

so we need to consider using script to copy config files when (un)installing.

File: bucket/rtss.json [hunk 1]
```diff
@@ -10,10 +10,13 @@
...
     "hash": "d2df6f2456040e66dc197f5626726d7ecd8870d353253ded147cb6648218322a",
     "pre_install": [
+        "ensure \"$persist_dir\" | Out-Null",
         "Expand-7zipArchive \"$dir\\RTSSSetup*.exe\" -Removal",
         "Remove-Item \"$dir\\`$*\", \"$dir\\Guru3D.com\", \"$dir\\Uninstall*\" -Recurse",
         "Move-Item \"$dir\\RTSSHooks.dll.copy\" \"$dir\\RTSSHooks.dll\"",
-        "Move-Item \"$dir\\RTSSHooks64.dll.copy\" \"$dir\\RTSSHooks64.dll\""
+        "Move-Item \"$dir\\RTSSHooks64.dll.copy\" \"$dir\\RTSSHooks64.dll\"",
+        "'HotkeyHandler.cfg', 'OverlayEditor.cfg' | ForEach-Object {if (!(Test-Path \"$persist_dir\\Plugins\\Client\\$_\")) {New-Item \"$dir\\Plugins\\Client\\$_\" | Out-Null}}",
+        "Copy-Item \"$persist_dir\\Plugins\\Client\" \"$dir\\Plugins\\Client\" -Recurse -ErrorAction 'SilentlyContinue'"
```
Comment (HUMORCE) [inline bucket/rtss.json:19]:
```suggestion
        "# configs",
        "Copy-Item \"$persist_dir\\Plugins\\Client\\*.cfg\" \"$dir\\Plugins\\Client\" -ErrorAction SilentlyContinue"
```

File: bucket/rtss.json [hunk 2]
```diff
@@ -30,10 +33,8 @@
             "RivaTuner Statistics Server"
         ]
     ],
-    "persist": [
-        "Plugins\\Client\\Overlays",
-        "Profiles"
-    ],
+    "persist": "Profiles",
+    "pre_uninstall": "if (Test-Path \"$persist_dir\") {Copy-Item \"$dir\\Plugins\\Client\" \"$persist_dir\\Plugins\\Client\" -Recurse -ErrorAction 'SilentlyContinue'}",
```
Comment (HUMORCE) [inline bucket/rtss.json:37]:
```suggestion
    "pre_uninstall": [
        "# persist configs",
        "Get-ChildItem \"$dir\\Plugins\\Client\" -Filter '*.cfg' | Copy-Item -Destination \"$persist_dir\\Plugins\\Client\" -Force -ErrorAction SilentlyContinue"
    ],
```

File: bucket/rtss.json [hunk 3]
```diff
@@ -30,10 +33,8 @@
             "RivaTuner Statistics Server"
         ]
     ],
-    "persist": [
-        "Plugins\\Client\\Overlays",
-        "Profiles"
-    ],
+    "persist": "Profiles",
```
Comment (HUMORCE) [inline bucket/rtss.json:36]:
```suggestion
    "persist": [
        "Plugins\\Client\\Overlays",
        "Profiles"
    ],
```

File: bucket/rtss.json [hunk 4]
```diff
@@ -10,10 +10,13 @@
     "url": "https://ftp.nluug.nl/pub/games/PC/guru3d/afterburner/%5BGuru3D.com%5D-RTSS.zip",
     "hash": "d2df6f2456040e66dc197f5626726d7ecd8870d353253ded147cb6648218322a",
     "pre_install": [
+        "ensure \"$persist_dir\" | Out-Null",
```
Comment (HUMORCE) [inline bucket/rtss.json:13]:
```suggestion
```

**Review by HUMORCE — APPROVED** [review]

**Pull request merged**